script/copy_deduplicate

```diff
@@ -213,11 +214,17 @@ def get_temporary_table(client, date):
     "_" and table_id starts with "anon" and is followed by at least one
     character.
     """
-    return get_temporary_dataset(client).table(f"anon{uuid4().hex}${date:%Y%m%d}")
+    dataset = get_temporary_dataset(client)
+    table = bigquery.Table(dataset.table(f"anon{uuid4().hex}"), schema)
+    table.expires = datetime.now() + timedelta(days=1)
+    table.clustering_fields = clustering_fields
+    table.time_partitioning = time_partitioning
+    table = client.create_table(table, exists_ok=False)
+    return f"{sql_full_table_id(table)}${date:%Y%m%d}"
 
 
 def sql_full_table_id(table):
-    return table.full_table_id.replace(":", ".")
+    return f"{table.project}.{table.dataset_id}.{table.table_id}"
```

Comment:
`table.full_table_id` is only available when set by the server on `Table` and `TableListItem` objects, so I updated this function correctly handle `TableReference` found in `query_job.destination` rather than make extra api calls.